### ISSUE TITLE
tintColor not applied to CastButton initially on Android

### ISSUE THREAD

**Issue by arsimr16** [react-native-google-cast/react-native-google-cast#315]
In my React code I'm trying to set the tintColor on the CastButton: `<CastButton style={tintColor='#ffffff'} />` 

I tried adding a `backgroundColor` to the `CastButton`.  This allowed me to see that the button is there, but the icon was black on a black background so I couldn't see it.  After pressing the button and starting a cast session, the tintColor is applied to the button.  I tried `<CastButton tintColor={'#ffffff'} />` which didn't work either.  I also tried a few different format values for the color since I noticed that in the Java code, the color type is an integer. 

I took a look at `RNGoogleCastButtonManager.java` and first tried things like hardcoding `mColor` to some value instead of null, manually calling `setTintColor` from `MediaRouteButton createViewInstance`, and calling `applyTint` from the `ColorableMediaRouteButton` constructor.  I added some logging and realized that I was getting inside `applyTint` but it was exiting early because `mRemoteIndicatorDrawable` was `null`.  This is supposed to be set inside `setRemoteIndicatorDrawable` but this doesn't seem to be called until after starting a cast session.  I saw some other threads about customizing the `CastButton` online which involved getting a `Drawable d` and doing something similar to what should happen in `applyTint`.  I never ended up getting this to work, but I'm also not very confident in my Java knowledge so I could be missing something simple.

I eventually found this documentation for customizing the cast button by creating a custom theme: https://developers.google.com/cast/docs/android_sender/customize_ui  I ended up getting this to work, so posting this in case it's helpful to anyone else:

#### react/android/app/src/main/res/values/styles.xml
```
<style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
    <item name="mediaRouteTheme">@style/CustomMediaRouterTheme</item>
</style>
<style name="CustomMediaRouterTheme" parent="Theme.MediaRouter.Light">
    <item name="mediaRouteButtonStyle">@style/CustomMediaRouteButtonStyle</item>
</style>
<style name="CustomMediaRouteButtonStyle" parent="Widget.MediaRouter.MediaRouteButton">
    <item name="mediaRouteButtonTint">#ffffff</item>
</style>
```

For now this workaround is fine, but is there some way to set `mRemoteIndicatorDrawable` from the constructor or soon after that and then to call `applyTint` so the custom theme is not necessary?

**Comment by Cartas** [comment]
I cannot for the life of me get the colour to change.  I've tried setting it in `styles.xml` as well as making changes in the Java code itself.

How did you get the `styles.xml` changes to actually work?  :cold_sweat:

**Comment by arsimr16** [comment]
I originally didn't see the changes because I was editing react/android/app/src/**main**/res/values/styles.xml but there was another file react/android/app/src/**debug**/res/values/styles.xml which was overriding the file I was editing.  I ended up deleting the debug version since I didn't have any styles that were debug-only, and then I saw the changes from the main version.  Any chance the same thing is happening for you?

**Comment by Cartas** [comment]
Afraid not, but I managed to get it working with the following Java change - in case it is of use to anyone:

```patch
diff --git a/node_modules/react-native-google-cast/android/src/main/java/com/reactnative/googlecast/components/RNGoogleCastButtonManager.java b/node_modules/react-native-google-cast/android/src/main/java/com/reactnative/googlecast/components/RNGoogleCastButtonManager.java
index eed490e..288444b 100644
--- a/node_modules/react-native-google-cast/android/src/main/java/com/reactnative/googlecast/components/RNGoogleCastButtonManager.java
+++ b/node_modules/react-native-google-cast/android/src/main/java/com/reactnative/googlecast/components/RNGoogleCastButtonManager.java
@@ -1,8 +1,10 @@
 package com.reactnative.googlecast.components;

 import android.content.Context;
+import android.content.res.TypedArray;
 import android.graphics.drawable.Drawable;
 import android.util.AttributeSet;
+import android.view.ContextThemeWrapper;
 import android.view.View;

 import androidx.core.graphics.drawable.DrawableCompat;
@@ -16,7 +18,6 @@ import com.google.android.gms.cast.framework.CastContext;
 import com.google.android.gms.cast.framework.CastState;
 import com.google.android.gms.cast.framework.CastStateListener;

-
 import java.util.ArrayList;
 import java.util.List;

@@ -43,10 +44,22 @@ public class RNGoogleCastButtonManager
     CastContext castContext = CastContext.getSharedInstance(context);

     final MediaRouteButton button = new ColorableMediaRouteButton(context);
+
+    Context otherContext = new ContextThemeWrapper(context, androidx.mediarouter.R.style.Theme_MediaRouter);
+
+    Drawable drawable = null;
+    TypedArray a = otherContext.obtainStyledAttributes(null,
+            androidx.mediarouter.R.styleable.MediaRouteButton, androidx.mediarouter.R.attr.mediaRouteButtonStyle, 0);
+    drawable = a.getDrawable(
+            androidx.mediarouter.R.styleable.MediaRouteButton_externalRouteEnabledDrawable);
+    a.recycle();
+
     CastButtonFactory.setUpMediaRouteButton(context, button);

     updateButtonState(button, castContext.getCastState());

+    button.setRemoteIndicatorDrawable(drawable);
+
     castContext.addCastStateListener(new CastStateListener() {
       @Override
       public void onCastStateChanged(int newState) {
```

**Comment by demianp** [comment]
@goncalo-assuncao for me it started happening because I didnt configure castFrameworkVersion = "19.0.0"
So, it was taking the last release 20.0.0 which was published on april 20.

**Comment by redefinered** [comment]
@demianp that's the solution that worked. Thanks!

**Comment by yvbeek** [comment]
@arsimr16 Could you please turn that fix into a PR?

**Comment by arsimr16** [comment]
My fix was to create a custom theme in the styles file.  Everyone would not want the same style as me.  I think the Java fix by @Cartas is a better candidate for a PR.

**Comment by yvbeek** [comment]
@arsimr16 Sorry, I pinged the wrong person

I'm creating a PR for @Cartas 's change

**Comment by petrbela** [comment]
This should now be fixed in v4.1.0. Feel free to open another issue if you're still experiencing problems.
